### DISCUSSION TITLE
Why choose dind configuration for self-hosted Renovate on Gitlab?

### DISCUSSION THREAD

**Discussion by catostrophe** [renovatebot/renovate#13209] [Request Help] [answered]
I am trying to configure a self-hosted Renovate instance on our company's private GitLab. We have docker runners, which I cannot configure myself.

I am choosing between a slim Renovate image + dind versus a fat Renovate image. What exact Renovate features require the slim+dind configuration? Can I stay on the fat image without losing anything? I cannot try dind yet because I can't set the privileged flag and our self-signed certs seem to need some extra configuration.

**Comment by catostrophe** [comment, marked as answer]
cc @viceice

**Reply by viceice** [reply]
Renovate can't respect tool version constraints in fat image, as there is only one version of each installed.

You should install a special runner for renovate only.

**Reply by viceice** [reply]
You can also use a rootless user docker install on a virtual machine. Renovate doesn't need any root privilege

**Reply by catostrophe** [reply]
Is a dedicated runner a must for running Renovate? What if you use any of the shared docker runners?

**Reply by catostrophe** [reply]
In wich case do I need to configure certs for the dind setup? Our gitlab host has a self-signed cert. Is it the case?

**Reply by viceice** [reply]
You can use shared runner of cause, but it needs dind support.

Custom certs is a little more complex, I'm sure there are some issues / discussions about it in this repo

**Reply by viceice** [reply]
Custom certs basics:
- https://github.com/containerbase/buildpack/blob/main/docs/custom-root-ca.md
- https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables

Cert should be there in renovate cache dir to be mounted to docker sidecar containers.

**Reply by viceice** [reply]
Another option is to build custom docker fat images, so they have matching tool versions installed.

We also have a new `binarySource=install` mode, but that currently only supports a few tools.

**Reply by catostrophe** [reply]
Thanks!

**Reply by kitos9112** [reply]
Thanks for having this discussion! It helped me to fix a similar issue with GL runners that cannot access the Docker socket, or cannot run dind.
